lib/SILGen/ManagedValue.h

```diff
@@ -219,20 +219,29 @@ class ManagedValue {
 
   /// Returns true if this managed value can be consumed.
   ///
-  /// This is true if either this value has a cleanup or if it is an
-  /// SSA value without ownership.
+  /// This is true if either this value has a cleanup or if it is a trivial
+  /// object value. For address values, this returns true only if the value has
+  /// a cleanup regardless of whether the type is trivial.
   ///
-  /// When an SSA value does not have ownership, it can be used by a consuming
-  /// operation without destroying it. Consuming a value by address, however,
-  /// deinitializes the memory regardless of whether the value has ownership.
+  /// When an object value is trivial, it can be passed to a consuming operation
+  /// without destroying it. Consuming a value by address, however, always
+  /// deinitializes the memory regardless of its type.
+  ///
+  /// Use this before emitting an operation that "takes" this vaule or passing
+  /// this value to a call that consumes the argument.
   bool isPlusOne(SILGenFunction &SGF) const;
 
   /// Returns true if this managed value can be forwarded without necessarilly
   /// destroying the original.
   ///
-  /// This is true if either isPlusOne is true or the value is trivial. A
-  /// trivial value in memory can be forwarded as a +1 value without
-  /// deinitializing the memory.
+  /// This is true if either isPlusOne is true or the value is trivial. Unlike
+  /// isPlusOne(), this returns true for trivial addres values. A +1 value can
```

Comment:
addres -> address.

I would also say this returns true for a trivial address value regardless of whether or not it has a cleanup attached.